manager/src/main/java/com/redhat/service/smartevents/manager/workers/WorkManagerImpl.java

```diff
@@ -2,124 +2,87 @@
-        Work w = workDAO.findById(work.getId());
-        w.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
-        w.setAttempts(w.getAttempts() + 1);
-    }
-
-    @Override
-    @Transactional
-    public void complete(Work work) {
-        if (!exists(work)) {
-            return;
-        }
-        // Work has been serialised by VertX at this point and has therefore lost all affinity with
-        // a JPA session. We therefore need to delete it by Id and not the entity itself.
-        workDAO.deleteById(work.getId());
-    }
-
-    @SuppressWarnings("unused")
-    @Scheduled(cron = "{event-bridge.resources.workers.schedule}", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
-    protected void processWorkQueue() {
-        processWorkQueue(getWorkQueue());
+    public void schedule(Processor processor) {
+        doSchedule(processor.getId(), processorWorkerJob);
     }
 
-    @Transactional
-    protected List<Work> getWorkQueue() {
-        return workDAO.findByWorkerId(workerId);
-    }
-
-    @Transactional(Transactional.TxType.NEVER)
-    protected void processWorkQueue(List<Work> work) {
-        work.forEach(this::fireEvent);
-    }
+    // Schedule the first execution of a Job.
+    // We cannot use Quartz scheduler to invoke the job repeatedly as we store mutable state within the
+    // Trigger definition and Quartz does not support updating the Trigger JobDataMap after the Trigger
+    // has been created. Rescheduling of the Job is handled by AbstractWorker if either an execution
+    // failed or the Work remained incomplete.
+    protected void doSchedule(String id, JobDetail jobDetail) {
+        try {
+            TriggerKey key = new TriggerKey(id, MANAGED_RESOURCES_GROUP);
+            if (Objects.nonNull(quartz.getTrigger(key))) {
+                LOGGER.info(String.format("Work already exists for '%s'. Skipping.", id));
+            }
 
-    @SuppressWarnings("unused")
-    @Transactional
-    @Scheduled(every = "5m", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
-    void reconnectDroppedWorkers() {
-        ZonedDateTime age = ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(5);
-        workDAO.reconnectDroppedWorkers(workerId, age);
+            // We have to store the serialised JobData due to limitations in Quarkus. See WorkManager for details.
+            Trigger trigger = TriggerBuilder.newTrigger()
+                    .forJob(jobDetail)
+                    .withIdentity(key)
+                    .usingJobData(STATE_FIELD_ID, id)
+                    .usingJobData(STATE_FIELD_ATTEMPTS, String.valueOf(0))
+                    .usingJobData(STATE_FIELD_SUBMITTED_AT, ZonedDateTime.now(ZoneOffset.UTC).toString())
+                    .startNow()
+                    .build();
+            quartz.scheduleJob(trigger);
+
+        } catch (SchedulerException e) {
+            LOGGER.info(String.format("Unable to schedule work for '%s'", id), e);
```

Comment:
@manstis I think we need to throw this `Exception`. If we don't then the associated resource will be forever stuck in the `ACCEPTED` status. By throwing the `Exception`, we can rollback the DB transaction commit which will prevent the resource from being created (or updated).